database/availabilities.py

```diff
@@ -0,0 +1,57 @@
+from bson import ObjectId
+
+from database import collection_availabilities
+from models.models import Availability
+
+"""
+db integrity:
+- schedule delete - delete all schedule's availabilities
+- operator delete - delete all operator's availabilities
+- add availability - check for no duplications - NOT CHECKING
+"""
+
+
+# Schedule and operator ids are optional
+def get_availabilities(schedule_id: str, operator_id: str):
+    if schedule_id:
+        if operator_id:
+            # schedule and operator
+            result = collection_availabilities.find(
+                {"schedule_id": schedule_id, "operator_id": operator_id})
+        else:
+            # only schedule
+            result = collection_availabilities.find({"schedule_id": schedule_id})
+    elif operator_id:
+        # only operator
+        result = collection_availabilities.find({"operator_id": operator_id})
+    else:
+        # neither
+        result = collection_availabilities.find({})
+    return [convert_unserializable(res) for res in list(result)]
+
+
+def add_availability(availability: Availability):
+    return collection_availabilities.insert_one(availability.dict())
+
+
+def delete_availability(availability_id: str):
+    return collection_availabilities.delete_one({"_id": ObjectId(availability_id)})
+
+
+def delete_availabilities_after_deletion(schedule_id: str = "", operator_id: str = ""):
```

Comment:
why to assign a default value?

Reply:
this function is being called only when a schedule or operator is deleted, so one of them will have a value and the other will not

when deleting a schedule - calling it this way :
delete_requests_after_deletion(schedule_id, None)
is the same but i preferred putting default values